Sources/Fluent/Query/Join.swift

```diff
@@ -18,10 +18,34 @@ public struct Join {
     /// the local data
     public let foreign: Entity.Type
 
+    /// Direction the foreign key goes
+    public let child: ChildDirection
```

Comment:
I think maybe `childDirection` would be a little more clear. We're throwing around tons of terms in Fluent, I think we need to start opting for being a bit more explicit. I'd expect a child model